### PR TITLE
ETL syncer

### LINKED ISSUE #12659: ETL syncer service
Once we have created the ETL database and models, we will need a basic syncer service class structure that can take an origin model and target model and keep them in sync.

Example:

```ruby
class ETL::Syncer
  def initialize(origin:, target:)
    @origin = origin
    @target = target
  end

  def call
    instances_needing_update.each do |original|
      target.sync_with_original(original)
    end
  end

  private

  attr_reader :origin, :target 

  def instances_needing_update
    # looked at origin.`updated_at` vs Time.now e.g.
  end
end
```

## AC

- [x] Syncer classes for
  * [x] Organization
  * [x] Person
  * [x] OrganizationsUser
  * [x] User
  * [x] Appeal
  * [x] Task

### PR DESCRIPTION
**Pull request by pkarman**
connects #12659 

### Description
ETL models must be kept in sync with their corresponding original models. Ordinarily this is an incremental process (sync everything that has changed since the last time we synced). Sometimes we will need a full sync (as for the first time, or for schema/logic changes).

This PR introduces the `ETL::Syncer` service and its subclasses. Each ETL model has a corresponding Syncer class.

* The Syncer is in charge of determining what needs syncing and iterating over rows
* The Model is in charge of the sync mappings between models

*NOTE* that Appeal and Task ETL models are not in this PR. They will come in a follow-on PR.

### Testing Plan
1. Automated tests pass.
2. Run manually in UAT after deployment.

### Documentation Updates
- [x] Topline File Descriptions Added or Updated as Needed

### CODE REVIEW

File: Makefile.example [hunk 1]
```diff
@@ -95,7 +95,7 @@ c:  ## Start rails console
 	bundle exec rails console
 
 etl-migrate:	## Migrate ETL database
-	DB=etl bundle exec rake db:migrate
+	DB=etl bundle exec rake db:migrate db:test:prepare
```
Comment (pkarman) [inline Makefile.example]: apply migrations to test db

File: app/services/etl/syncer.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  def target_class
+    fail "Must override abstract method target_class"
+  end
+
+  private
+
+  attr_reader :since
+
+  def incremental?
```
Comment (lomaxap) [inline app/services/etl/syncer.rb]: what scenarios would this be incremental?

File: app/models/etl/record.rb [hunk 3]
```diff
@@ -5,4 +5,25 @@
 class ETL::Record < ApplicationRecord
   self.abstract_class = true
   establish_connection :"etl_#{Rails.env}"
+
+  class << self
+    def sync_with_original(original)
```
Comment (lomaxap) [inline app/models/etl/record.rb]: is there ever an instance where this wouldn't be overridden by the child class?

Comment (lomaxap) [reply app/models/etl/record.rb]: `organization_users`, `people`, and `organization` models?

File: app/services/etl/syncer.rb (hunk 2 above)
Comment (pkarman) [reply app/services/etl/syncer.rb]: I expect we'll do an incremental (hourly?) job to keep ETL updated throughout the week, maybe a full sync from scratch on the weekends.

**Pull request merged**